### DISCUSSION TITLE
Help designing schema

### DISCUSSION THREAD

**Discussion by brielov** [prisma/prisma#3326] [Q&A] [answered]
I have the following schema:

```prisma
model Transaction {
  source User
  destination User
  amount Float
}

model User {
  id String @id @default(cuid())
  name String

  purchases Transaction[] // references the source field on transaction
  sales Transaction[] // reeferences the destination field on transaction
}
```

How would you approach making this relationship work?

EDIT: No matter what I try it always warns me something like `Error validating: The argument 'references' must refer to a unique criteria in the related model 'Transaction'. But it is referencing the following fields that are not a unique criteria: sourceId`

**Comment by ryands17** [comment, marked as answer]
Hey @brielov 👋 
Could you try this:

```prisma
model Entity {
  id   String @id @default(cuid())
  name String

  purchases Transact[] @relation("purchases")
  sales     Transact[] @relation("sales")
}

model Transact {
  id            String   @id @default(cuid())
  source        Entity   @relation("purchases", fields: [sourceId], references: [id])
  destination   Entity   @relation("sales", fields: [destinationId], references: [id])
  amount        Float
  date          DateTime @default(now())
  sourceId      String
  destinationId String
}
```

Also in version _2.4.1_ of Prisma, `Transaction` is a reserved word, so you would need to change that accordingly. This should be fixed in a future release of Prisma.
Let me know if this works.

**Reply by brielov** [reply]
Awesome, thank you so much!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
